react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+
+export default function HotButton() {
+  let [clicks, setClicks] = useState(0);
+
+  function handleClick() {
+    setClicks(clicks += 1);
+  }
+
+  function createHotButton(className) {
```

Comment:
Interesting approach, but let's make it simpler and more idiomatic React.

Instead of using a function here, just return the JSX directly from the component code. Components usually have something like `return <button className=........>.......` as their last line.